api/src/pcapi/scripts/offerer/synchronize_venues_banners_with_google_places.py

```diff
@@ -0,0 +1,190 @@
+        .order_by(offerers_models.Venue.id)
+    )
+    if begin is not None:
+        query = query.filter(begin <= offerers_models.Venue.id)
+    if end is not None:
+        query = query.filter(offerers_models.Venue.id < end)
+    if limit is not None:
+        query = query.limit(limit)
+    venues = query.all()
+    return venues
+
+
+def get_place_id(name: str, address: str | None) -> str | None:
+    address = address or ""
+    result = FindPlaceResponse.model_validate(
+        gmaps.find_place(input=name + address, input_type="textquery", fields=["place_id"])
+    )
+    if result.status == "OK" and len(result.candidates):
+        return result.candidates[0].place_id
+    return None
+
+
+def is_the_owner(name: str, photo: PlacePhoto) -> bool:
+    for attr in photo.html_attributions:
+        if name in attr:
+            return True
+    return False
+
+
+def get_place_photos_and_owner(place_id: str) -> PlaceDetails | None:
+    result = PlaceResponse.model_validate(gmaps.place(place_id=place_id, fields=["photo", "name"]))
+    if result.status != "OK":
+        return None
+    return result.result
+
+
+def get_owner_photo(photos: list[PlacePhoto], owner_name: str) -> PlacePhoto | None:
+    for photo in photos:
+        if is_the_owner(owner_name, photo):
+            return photo
+    return None
+
+
+storage_folder = settings.THUMBS_FOLDER_NAME + "/google_places"
+
+
+def save_photo_to_gcp(venue_id: int, photo: PlacePhoto, prefix: str) -> str:
+    photos_result = gmaps.places_photo(
+        photo_reference=photo.photo_reference,
+        max_width=photo.width,
+    )
+    object_id = f"google_places_{prefix}{venue_id}.jpeg"
+    data = bytes()
+    for chunk in photos_result:
+        if chunk:
+            data += chunk
+    store_public_object(
+        storage_folder,
+        object_id,
+        image_conversion.standardize_image(
+            content=data, ratio=image_conversion.ImageRatio.LANDSCAPE, crop_params=image_conversion.CropParams  # type: ignore [arg-type]
+        ),
+        "image/jpeg",
+    )
+    return settings.OBJECT_STORAGE_URL + "/" + storage_folder + "/" + object_id
+
+
+@blueprint.cli.command("synchronize_venues_banners_with_google_places")
+@click.argument("begin", type=int, required=True)
+@click.argument("end", type=int, required=True)
+@click.argument("limit", type=int, required=False, default=None)
+def synchronize_venues_banners_with_google_places(begin: int, end: int, limit: int | None = None) -> None:
+    venues_without_photos = get_venues_without_photo(begin, end, limit)
+    print("the number of venues is:", len(venues_without_photos))
+    logger.info(
+        "[gmaps_banner_synchro] synchronize_venues_banners_with_google_places %s %s %s: the number of venues tried: %s",
+        begin,
+        end,
+        limit,
+        len(venues_without_photos),
+    )
+    number_of_places_found = 0
+    number_places_with_at_least_one_photo = 0
+    number_of_places_with_owner_photo = 0
```

Comment:
[détail / nommage]

Remarques objectives : 

- "number_places" ou "number_**of**_places" ? Il faut choisir. :)

Remarques probablement plus subjectives : 
- est-ce que le suffixe "_found" apporte quelque chose ?
- propositions : `nb_places`, `nb_places_with_photo`, `nb_places_with_owner_photo`.

Reply:
pour le found je pense que oui. parcequ' il y' a le nombre de venues qu'on a cherché et le nombre de venues qu'on a trouvé sur places api. je pense que ça apporte en lisibilité 
pour le reste je corrige